src/components/forms/transaction/TransactionForm.tsx

```diff
@@ -113,11 +114,14 @@ async function onSubmit(data: FormValues, action: 'add' | 'update' | 'delete', o
     date: DateTime.fromISO(data.date),
   });
 
-  if (action === 'add' || action === 'update') {
+  if (action === 'add') {
     await transaction.save();
-  }
-
-  if (action === 'delete') {
+  } else if (action === 'update') {
+    await transaction.save();
+    await Split.delete({
+      fk_transaction: IsNull(),
```

Comment:
I guess we could be more specific and search for the split guid difference but.. I can't think of an edge case where this is bad and as an extra we get a sanity check (we don't want splits with null transaction).